oras/provider.py

```diff
@@ -596,24 +611,27 @@ def chunked_upload(
         # Read the blob in chunks, for each do a patch
         start = 0
         with open(blob, "rb") as fd:
-            for chunk in oras.utils.read_in_chunks(fd):
-                if not chunk:
```

Comment:
Why did you remove this check?

Reply:
Even if your iterator doesn't raise, it stops yielding after a certain point, and there's an internal check with the same condition too. Why should it be necessary to check here again?

Reply:
You're right - I forgot that I added that - it must have not been there at some point. Reference:

https://github.com/oras-project/oras-py/blob/caf8db5b279382335fbb1f6d7402ed9b73618d37/oras/utils/fileio.py#L260-L273